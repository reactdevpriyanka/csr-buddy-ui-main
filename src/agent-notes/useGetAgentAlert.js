import axios from 'axios';
import useSWR from 'swr';
import useSanitizedRouter from '@/hooks/useSanitizedRouter';

export default function useGetAgentAlert() {
  const { id } = useSanitizedRouter();

  return useSWR(`/api/v1/agentNotes?customerId=${id}`, async (url) =>
    axios.get(url).then(({ data }) => data.map((alerts) => ({ ...alerts }))),
  );
}
