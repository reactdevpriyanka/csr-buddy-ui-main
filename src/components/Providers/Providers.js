import PropTypes from 'prop-types';
import DateFnsUtils from '@date-io/date-fns';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import { ThemeProvider } from '@material-ui/core/styles';
import { Slide } from '@material-ui/core';
import { SnackbarProvider } from 'notistack';
import theme from '@/theme';
import SnackMessage from '@components/SnackMessage';
import RightModalProvider from './RightModalProvider';
import NavigationProvider from './NavigationProvider';
import BreadcrumbProvider from './BreadcrumbProvider';

const Providers = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <SnackbarProvider
        autoHideDuration={3000}
        anchorOrigin={{
          vertical: 'top',
          horizontal: 'center',
        }}
        TransitionComponent={Slide}
        content={(key, { messageHeader, messageSubheader, listItems, variant, link }) => (
          <SnackMessage
            variant={variant}
            id={key}
            messageHeader={messageHeader}
            messageSubheader={messageSubheader}
            listItems={listItems}
            link={link}
          />
        )}
      >
        <MuiPickersUtilsProvider utils={DateFnsUtils}>
          <BreadcrumbProvider>
            <NavigationProvider>
              <RightModalProvider>{children}</RightModalProvider>
            </NavigationProvider>
          </BreadcrumbProvider>
        </MuiPickersUtilsProvider>
      </SnackbarProvider>
    </ThemeProvider>
  );
};

Providers.propTypes = {
  children: PropTypes.node,
};

export default Providers;
