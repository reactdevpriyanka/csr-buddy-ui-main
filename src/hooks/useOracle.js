export default function useOracle() {
  if (typeof window !== 'undefined') {
    return window.__ORACLE__;
  }
}
