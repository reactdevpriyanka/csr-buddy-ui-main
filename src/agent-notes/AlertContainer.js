import { useState, useCallback, useMemo } from 'react';
import { Grid, makeStyles } from '@material-ui/core';
import Box from '@mui/material/Box';
import { useFeature } from '@/features';
import useAthena from '@/hooks/useAthena';
import { useAlertSummary } from '@/hooks/useAlertSummary';
import ConnectedSideNavCards from '@/components/Base/ConnectedSideNavCards/ConnectedSideNavCards';
import CreateAlertButton from './CreateAlertButton';
import useGetAgentAlert from './useGetAgentAlert';
import CreateAlertForm from './CreateAlertForm';
import MarkAsRead from './MarkAsRead';

const useStyles = makeStyles((theme) => ({
  scrollContainer: {
    padding: theme.utils.fromPx(24),
    height: '100%',
    overflowY: 'auto',
  },
  noAlerts: {
    marginTop: theme.utils.fromPx(10),
    marginBotton: theme.utils.fromPx(24),
    textAlign: 'center',
  },
  alerts: {
    margin: `${theme.utils.fromPx(10)} 0px ${theme.utils.fromPx(24)} 0px`,
    textAlign: 'center',
    color: '#602B00',
  },
}));

const AlertContainer = () => {
  const classes = useStyles();
  const { getLang } = useAthena();
  const createAgentAlertBtnEnabled = useFeature('feature.explorer.createAgentAlertBtnEnabled');
  const [showCreateAlertForm, setShowCreateAlertForm] = useState(false);

  const { data: agentAlerts = [], mutate: mutateGetAgentAlert } = useGetAgentAlert();
  const alertDataValues = useMemo(() => Object.values(agentAlerts));
  const { mutate: mutateAlertSummary } = useAlertSummary();

  const onAfterCreateAlert = useCallback(() => {
    setShowCreateAlertForm(false);
    mutateGetAgentAlert();
    mutateAlertSummary();
  }, [mutateAlertSummary, mutateGetAgentAlert, setShowCreateAlertForm]);

  const handleMarkAsRead = useCallback(() => {
    mutateGetAgentAlert();
    mutateAlertSummary();
  }, [mutateAlertSummary, mutateGetAgentAlert]);

  return (
    <div className={classes.scrollContainer}>
      <Grid>
        {showCreateAlertForm ? (
          <CreateAlertForm
            onAfterSubmit={onAfterCreateAlert}
            onClose={() => setShowCreateAlertForm(false)}
          />
        ) : (
          <>
            {createAgentAlertBtnEnabled && (
              <form>
                <Box sx={{ display: 'flex', justifyContent: 'center' }}>
                  <CreateAlertButton onClick={() => setShowCreateAlertForm(true)} />
                </Box>
                <p className={classes.alerts} data-testid="alert:agent alert maximum">
                  {getLang('maximumAgentAlertText', { fallback: '2 alerts per agent maximum' })}
                </p>
              </form>
            )}
            {alertDataValues?.length > 0 ? (
              <>
                {alertDataValues.map((alert) => (
                  <ConnectedSideNavCards key={alert.id}>
                    <MarkAsRead alertData={alert} onMarkAsRead={handleMarkAsRead} />
                  </ConnectedSideNavCards>
                ))}
                <p className={classes.noAlerts} data-testid="alert:end of alerts">
                  {getLang('endAgentAlertText', { fallback: 'End of Alerts' })}
                </p>
              </>
            ) : (
              <p className={classes.noAlerts} data-testid="alert:no agent alert">
                {getLang('noAgentAlertText', { fallback: 'No Agent Alerts' })}
              </p>
            )}
          </>
        )}
      </Grid>
    </div>
  );
};

export default AlertContainer;
