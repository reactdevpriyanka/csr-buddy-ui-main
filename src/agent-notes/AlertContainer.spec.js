import { screen } from '@testing-library/dom';
import { render, fireEvent } from '@testing-library/react';
import { useRouter } from 'next/router';
import useOracle from '@/hooks/useOracle';
import useGetAgentAlert from '@/agent-notes/useGetAgentAlert';
import Providers from '@components/Providers';
import AlertContainer from './AlertContainer';

jest.mock('next/router', () => ({
  useRouter: jest.fn(),
}));

jest.mock('@/hooks/useOracle');
jest.mock('@/agent-notes/useGetAgentAlert');

describe('<AlertContainer />', () => {
  beforeAll(() => {
    useRouter.mockReturnValue({
      query: {
        id: '5555',
      },
    });

    useOracle.mockReturnValue({
      getIncidentStartData: () => ({ interactionId: '9999' }),
      emit: jest.fn(),
    });
  });

  test('it should display Create Agent Alert form', () => {
    useGetAgentAlert.mockReturnValue({
      agentAlerts: [],
    });

    render(
      <Providers>
        <AlertContainer />
      </Providers>,
    );

    const createAlertBtn = screen.getByTestId('alert:create agent alert');
    expect(createAlertBtn).toBeInTheDocument();
    fireEvent.click(createAlertBtn);

    expect(render()).toMatchSnapshot();

    const form = screen.getByTestId('create-agent-alert');
    expect(form).toBeInTheDocument();

    const title = screen.getByTestId('alert:agent alert title');
    expect(title).toBeInTheDocument();
    const description = screen.getByTestId('alert:agent alert paragraph');
    expect(description).toBeInTheDocument();

    const alertTypeInput = screen.getByTestId('add-alert-type');
    expect(alertTypeInput).toBeInTheDocument();
    const alertNote = screen.getByTestId('add-alert-note');
    expect(alertNote).toBeInTheDocument();

    const submitBtn = screen.getByTestId('save-agent-alert');
    expect(submitBtn).toBeInTheDocument();
    const cancelBtn = screen.getByTestId('add-agent-alert-cancel');
    expect(cancelBtn).toBeInTheDocument();
  });
});
