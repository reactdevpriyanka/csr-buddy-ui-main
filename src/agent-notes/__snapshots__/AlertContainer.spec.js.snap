// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertContainer /> it should display Create Agent Alert form 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-scrollContainer-1"
      >
        <div
          class="MuiGrid-root"
        >
          <form
            class="makeStyles-root-6"
            data-testid="create-agent-alert"
          >
            <h2
              class="makeStyles-titleLabel-7"
              data-testid="alert:agent alert title"
            >
              Create Agent Alert
            </h2>
            <p
              data-testid="alert:agent alert paragraph"
            >
              Agent Alerts are for passing 
              <span
                style="font-weight: bold;"
              >
                critical
              </span>
               customer or order information to the next agent(s).
            </p>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiFormControl-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root makeStyles-inputLabel-8 MuiInputLabel-formControl MuiInputLabel-animated"
                  data-shrink="false"
                  id="agent-type"
                >
                  Alert Type
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root makeStyles-dropdown-9 MuiInputBase-formControl"
                  data-testid="add-alert-type"
                >
                  <div
                    aria-haspopup="listbox"
                    aria-labelledby="agent-type"
                    class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                    role="button"
                    tabindex="0"
                  >
                    <span>
                      â€‹
                    </span>
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput"
                    tabindex="-1"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-15 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-17"
                    >
                      <span>
                        Alert Type
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="MuiFormControl-root"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root makeStyles-textField-10"
                  data-testid="add-alert-note"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                    data-shrink="false"
                  >
                    Agent Alert
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
                  >
                    <textarea
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                      maxlength="280"
                      rows="1"
                      style="height: 0px; overflow: hidden;"
                    />
                    <textarea
                      aria-hidden="true"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                      readonly=""
                      style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
                      tabindex="-1"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-15 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-17"
                      >
                        <span>
                          Agent Alert
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                  <p
                    class="MuiFormHelperText-root MuiFormHelperText-contained"
                  >
                    <div
                      class="makeStyles-helperText-12"
                    >
                      <span>
                        Note: 2 alerts per agent maximum
                      </span>
                      <span>
                        0/280
                      </span>
                    </div>
                  </p>
                </div>
              </div>
            </div>
          </form>
          <div
            class="makeStyles-buttonGrid-11"
          >
            <button
              aria-label="Cancel"
              class="MuiButtonBase-root MuiButton-root makeStyles-root-5 MuiButton-text makeStyles-cancelButton-14"
              data-testid="add-agent-alert-cancel"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Cancel
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="Save Alert"
              class="MuiButtonBase-root MuiButton-root makeStyles-root-5 MuiButton-text solid disabled makeStyles-submitButton-13 Mui-disabled Mui-disabled"
              data-testid="save-agent-alert"
              disabled=""
              tabindex="-1"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Save Alert
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
